### ISSUE TITLE
Missing table of contents for docmentation

### ISSUE THREAD

**Issue by apbarrero** [google/googletest#676]
I am a heavy user of Google Test documentation reference and have always promoted its use among colleagues claiming its extensive and complete documentation as a key feature.

Since the project home has been moved to Github, the table of contents for the docs has been lost. It was much easier to navigate through the docs before.

I guess Github's markdown can't automatically generate the TOC and keeping it hardcoded would be a maintenance pain and eventually would lead to inconsistent contents.

Would moving/linking the docs to the project Wiki help?

**Comment by CJCombrink** [comment]
I agree with @apbarrero, since the TOC is missing it is much harder to navigate the documentation.

Wouldn't it be possible to add a TOC to the Wiki pages of github and perhaps move the documentation there...

**Comment by apbarrero** [comment]
@BillyDonahue, are there any plans regarding the docs that you could share here?

I can help to moving the docs to the wiki if you find that a good solution.

**Comment by BillyDonahue** [comment]
Moving to wiki is not my favorite way to go. I like having the docs checked in as part of the project.
Perhaps GitHub has or is developing a way to generate a ToC from Markdown?

**Comment by apbarrero** [comment]
Just found [this answer is SO](http://stackoverflow.com/a/18244520/198011), and there are no references to ToC in [github's markdown reference](https://guides.github.com/features/mastering-markdown/).

I'm not confident that they might be implementing ToC automatic generation any soon.

**Comment by AlekseiCherkes** [comment]
+1

I think, It's not a big burden to generate the first version of the TOC by script and then update it manually.

[https://github.com/google/googletest/pull/646](url)

**Comment by Farwaykorse** [comment]
These are the main reference documents.
They are way too long and unorganized. You have to go through all to content to find a (potential) relevant section. And remember your way around.

I prefer a good Markdown documentation over the wiki, since it comes with the code and stays in sync with the code version you are using.  
Adding a TOC is essential to navigate these documents. But in-document linking is not a feature. Headers don't add have the option to add the required id tags.

#646 relies on GitHub making headers accessible by linking. Not entirely portable, but it is at least readable as plain-text.

edit: added a suggestion to #646 to make it portable.
